test/app/App/Panel/AddToken/index.test.js

```diff
@@ -7,162 +7,179 @@ import store from '../../../../../main/store'
-    await waitFor(() => expect(tokenNameInput.value).toEqual('Frame Test on Polygon'), { timeout: 200 })
-    expect(tokenSymbolInput.value).toEqual('mFRT')
-    expect(tokenDecimalsInput.value).toEqual('18')
+  
+    expect(link.send).toHaveBeenCalledWith('tray:action', 'navDash', {
+      view: 'tokens',
+      data: {
+        notify: 'addToken',
+        notifyData: {
+          chainId: 137,
+          address: '0x3432b6a60d23ca0dfca7761b7ab56459d9c964d0'
+        }
+      }
+    })
   })
 
-  it('should show a form with defaults when data is not found', async () => {
-    const { user, getByLabelText, getByRole, rerender } = setup(
-      <AddToken activeChains={[
-        { id: 1, name: 'Mainnet', connection: { primary: { connected: true } } },
-        { id: 137, name: 'Polygon', connection: { primary: { connected: true } } }
-      ]}
-      data={{ notifyData: { chainId: 137 }}} />, { advanceTimers }
-    )
+  it('should retrieve token metadata from a connected chain when an address is entered', async () => {
+    store.setPrimary('ethereum', 137, { connected: true })
 
-    rerenderComponent = rerender
+    const { user, getByLabelText, getByRole } = setup(<AddToken data={{ notifyData: { chainId: 137 }}} />)
 
     const contractAddressLabel = getByLabelText(`Enter token's address`)
-    await user.type(contractAddressLabel, '0x3432b6a60d23ca0dfca7761b7ab56459dinvalid')      
+    await user.type(contractAddressLabel, '0x3432b6a60d23ca0dfca7761b7ab56459d9c964d0')
     const setAddressButton = getByRole('button', { name: 'Set Address' })
     await user.click(setAddressButton)
+  
+    expect(link.invoke).toHaveBeenCalledWith('tray:getTokenDetails', '0x3432b6a60d23ca0dfca7761b7ab56459d9c964d0', 137)
+  })
+})
 
+describe('setting token details', () => {
+  it('should prompt for default token details', () => {
+    const { getByLabelText, getByRole } = render(
+      <AddToken data={{ notifyData: { chainId: 137, address: '0x64aa3364F17a4D01c6f1751Fd97C2BD3D7e7f1D4' }}} />
+    )
+  
     const contractAddressInput = getByRole('heading')
     const tokenNameInput = getByLabelText('Token Name')
     const tokenSymbolInput = getByLabelText('Symbol')
     const tokenDecimalsInput = getByLabelText('Decimals')
-
-    expect(contractAddressInput.textContent).toEqual('0x3432b6a6dinvalid')
+  
+    expect(contractAddressInput.textContent).toEqual('0x64aa3364D7e7f1D4')
     expect(tokenNameInput.value).toEqual('Token Name')
     expect(tokenSymbolInput.value).toEqual('SYMBOL')
     expect(tokenDecimalsInput.value).toEqual('?')
   })
+  
+  it('should update with loaded token metadata', async () => {
+    store.setPrimary('ethereum', 137, { connected: true })
+  
+    const { user, getByLabelText, getByRole, rerender } = setup(<AddToken data={{ notifyData: { chainId: 137 }}} />)
+  
+    link.invoke.mockResolvedValue({
+      name: 'Frame Test on Polygon',
+      symbol: 'mFRT',
+      decimals: 18
+    })
```

Comment:
I think we're missing a test for the unhappy path here - link invoke resolving with the "token not found" data should result in the default case

Reply:
I think this would just mean that invoke would return different data right? we have that part covered, as this test ensures that whatever data we get back, we display in the form

Reply:
Yes, but I would say you're still not testing the case where the data we get back is empty as that is a distinct case of the "token not found" response `{ name: '', symbol: '', decimals: 0 }` mapping to the default values as if the lookup had not occurred.

Reply:
does the fact that the values are the "default" values change anything? I don't see any special handling for this case. from the perspective of this component, we just need to make sure that we render the values we get back in the appropriate places. the content of the fields in the response isn't really relevant to testing this component.

Reply:
Perhaps not; I'd put the test in, you wouldn't, so maybe this is a preference thing.  Probably best to just merge it and be done as we've spent a lot of time on this component already.